### ISSUE TITLE
Review: MoreEnumerable.Subsets()

### ISSUE THREAD

**Issue by GoogleCodeExporter** [blyzer/morelinq#34]
```
Purpose of code changes on this branch:

Rev 117 - Implementation of a set of operators that produce a sequence of
subsets of an original source sequence. Subsets are produced in a deferred,
streaming manner. There are two primary overloads - one that produces all
subsets of a given sequence (of which there are 2^N), and an overload that
produces all k-sized subsets of a sequence (of which there are N!/((N-m)!m!)).

When reviewing my code changes, please focus on:

* The public interface of the extension methods.
* The effectiveness and clarity of the available XML comment documentation.
* How well these operators fit will into the MoreLINQ ecosystem of
extension methods.
* The extent of coverage available from the corresponding unit tests.
* The clarity and division of responsibility of the implementation.
* Whether edge cases are correctly identified and handled in the
implementation and tests.

After the review, I'll merge this branch into:
/trunk


```

Original issue reported on code.google.com by `ambientl...@gmail.com` on 17 Jan 2010 at 5:25

**Comment by GoogleCodeExporter** [comment]
Original comment by `azizatif` on 17 Jan 2010 at 11:22

**Comment by GoogleCodeExporter** [comment]
```
Now migrated to Hg and available in clone:
http://code.google.com/r/azizatif-morelinq-evenmore/
This clone is ready to be cloned :O) for further review. It also has the 
main/default branch already merged in to bring it up to date.
```

Original comment by `azizatif` on 25 May 2012 at 11:38
- Now blocked on: #37

**Comment by GoogleCodeExporter** [comment]
Original comment by `azizatif` on 11 Jun 2013 at 8:15
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Following no response from ambientlion (who possibly/assuming lost interest 
given it has taken years to get this far), taking ownership of issue to 
consider as addition for MoreLINQ 2.0 milestone.
```

Original comment by `azizatif` on 12 Jun 2013 at 10:10
- Added labels: **Component-Logic**, **Milestone-Release2.0**, **Priority-High**, **Type-Enhancement**
- Removed labels: **Priority-Medium**, **Type-Review**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision ecbb044b3f86.
```

Original comment by `azizatif` on 14 Jun 2013 at 5:18
- Changed state: **Fixed**
